frontend/src/metabase/core/components/Markdown/Markdown.styled.tsx

```diff
@@ -2,8 +2,9 @@ import { FC, ReactElement } from "react";
 import ReactMarkdown from "react-markdown";
 import styled from "@emotion/styled";
 import { color } from "metabase/lib/colors";
+import type { MarkdownProps } from "./Markdown";
 
-export const MarkdownRoot = styled(getComponent(ReactMarkdown))`
+export const MarkdownRoot = styled(getComponent(ReactMarkdown))<MarkdownProps>`
```

Comment:
this is not the best practice IMO. If you want to limit the number of props or some reason, you can do it of course with Pick helper, but I doubt it will make any difference.

overall you end up in the unnecessary interface duplication IMO

Reply:
after second look I agree with the link you shared - it's styled.input, which needs to know what kind of props are going to be provided - additional interface there makes total sense to me, but in my code it's `ReactMarkdown` - a component, which already has it's own interface

Reply:
Still, even if it's a component that already has its own interface it doesn't make any difference to me from say `styled.div` in the sense that each HTML element also has its own interface e.g. `onClick`, `aria-xxx`, and etc.

But in the styled-components files, the props would indicate *what* props we expect to use–and usually only exclusively–in the styled components. I think I've seen props that are used in both the components and in the styled-components, but that is rare e.g. `disabled`.

Anyway, this isn't something that we have baked into our styled guide imo, it's just a de facto practice I've seen we do. Feel free to discuss this with other engineers. I think creating a new interface specifically for a styled-component using `Pick` helper sounds promising as I don't want to be duplicate the type twice, the original goal is just to make an interface is clearly defines what props are used in styled-components. I think the `Pick` helper has an advantage that it's harder to create an interface whose properties don't exist in the original component's props. i.e. manually declaring a new interface is a bit prone to typos.